Mage.Sets/src/mage/sets/fallenempires/Necrite.java

```diff
@@ -0,0 +1,110 @@
+ */
+package mage.sets.fallenempires;
+
+import java.util.UUID;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.AttacksAndIsNotBlockedTriggeredAbility;
+import mage.abilities.costs.common.SacrificeSourceCost;
+import mage.abilities.effects.Effect;
+import mage.abilities.effects.common.DestroyTargetEffect;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.abilities.effects.common.continuous.AssignNoCombatDamageSourceEffect;
+import mage.cards.CardImpl;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.Rarity;
+import mage.filter.FilterPermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.target.Target;
+import mage.target.TargetPermanent;
+
+/**
+ *
+ * @author MarcoMarin
+ */
+public class Necrite extends CardImpl {
+
+    public Necrite(UUID ownerId) {
+        super(ownerId, 22, "Necrite", Rarity.COMMON, new CardType[]{CardType.CREATURE}, "{1}{B}{B}");
+        this.expansionSetCode = "FEM";
+        this.subtype.add("Thrull");
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Whenever Necrite attacks and isn't blocked, you may sacrifice it. If you do, destroy target creature defending player controls. It can't be regenerated.
+        DoIfCostPaid effect = new DoIfCostPaid(new DestroyTargetEffect(true), new SacrificeSourceCost(), "Sacrifice {this} to destroy target creature defending player controls?");
+        effect.setText("you may sacrifice it. If you do, destroy target creature defending player controls. It can't be regenerated.");
+        effect.addEffect(new AssignNoCombatDamageSourceEffect(Duration.EndOfTurn));
+        Ability ability = new NecriteTriggeredAbility(effect);
+        this.addAbility(ability);
+        
+    }
+
+    public Necrite(final Necrite card) {
+        super(card);
+    }
+
+    @Override
+    public Necrite copy() {
+        return new Necrite(this);
+    }
+}
+
+class NecriteTriggeredAbility extends AttacksAndIsNotBlockedTriggeredAbility{
+    
+    public NecriteTriggeredAbility(Effect effect) {
+        super(effect);
+    }
+
+    public NecriteTriggeredAbility(final NecriteTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public NecriteTriggeredAbility copy() {
+        return new NecriteTriggeredAbility(this);
+    }
+    
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        if (super.checkTrigger(event, game)){
+            UUID defendingPlayerId = game.getCombat().getDefendingPlayerId(getSourceId(), game);
+            FilterPermanent filter = new FilterCreaturePermanent();
+            filter.add(new ControllerIdPredicate(defendingPlayerId));
+            Target target = new TargetPermanent(filter);
+            this.addTarget(target);
```

Comment:
always clear the targets before you add targets during checkTrigger. Otherwise they accumulate and the ability has multiple targets over time.